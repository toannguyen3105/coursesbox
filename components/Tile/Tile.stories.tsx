import { ComponentMeta, ComponentStoryObj } from '@storybook/react'
import Tile from './Tile'

export default {
  title: 'Content/Tile',
  component: Tile,
} as ComponentMeta<typeof Tile>

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  args: {
    header: 'lorem ipsum dolor sit amet',
    children:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
  },
}
