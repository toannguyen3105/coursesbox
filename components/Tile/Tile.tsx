import styled from '@emotion/styled'
import { FC, ReactNode } from 'react'
import { borderRadius, boxShadow } from '@/components/styles'

type Props = {
  children: ReactNode
  header: string
}

const Section = styled.section`
  ${borderRadius};
  padding: 1vmin 4vmin 4vmin;
  background: ${({ theme }) => theme.background};
  color: ${({ theme }) => theme.font.regular};
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)}
`

const Tile: FC<Props> = ({ header, children }) => {
  return (
    <Section>
      <h2>{header}</h2>
      {children}
    </Section>
  )
}

export default Tile
