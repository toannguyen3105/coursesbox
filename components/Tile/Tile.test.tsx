import { render } from 'test-utils'
import Tile from './Tile'

describe('Tile test cases', () => {
  it('Tile render check', () => {
    const { asFragment } = render(
      <Tile header="Lorem ipsum dolor sit amet">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae,
        mollitia?
      </Tile>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
