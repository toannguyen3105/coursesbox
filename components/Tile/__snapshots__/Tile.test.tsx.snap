// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases Tile render check 1`] = `
<DocumentFragment>
  <section
    class="css-g8h8xw"
  >
    <h2>
      Lorem ipsum dolor sit amet
    </h2>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae, mollitia?
  </section>
</DocumentFragment>
`;
